ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -1,28 +1,45 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
-using ntbs_service.Data.Legacy;
+using ntbs_service.Services;
+using ntbs_service.Models;
+using Microsoft.EntityFrameworkCore;
 
 namespace ntbs_service.Pages_Search
 {
     public class IndexModel : PageModel
     {
-        private readonly ISearchService _searcher;
+        public INotificationService service;
+        public int PageIndex;
+        public string CurrentFilter { get; set; }
+        public PaginatedList<Notification> Notifications { get; set; }
+        public IList<NotificationBannerModel> SearchResultsBannerDisplay;
 
-        public IndexModel(ISearchService searcher)
+        public IndexModel(INotificationService service)
         {
-            _searcher = searcher;
-            Results = new List<SearchResult>();
+            this.service = service;
         }
 
-        [BindProperty]
-        public SearchRequest SearchRequest { get; set; }
+        public async Task<IActionResult> OnGetAsync(int? pageIndex)
+        {
 
-        public IList<SearchResult> Results { get; set; }
+            var pageSize = 50;
+
+            PageIndex = pageIndex ?? 1;
+
+            IQueryable<Notification> notificationsIQ = service.GetBaseNotificationIQueryable();
+
+            Notifications = await PaginatedList<Notification>.CreateAsync(
+                notificationsIQ.AsNoTracking(), pageIndex ?? 1, pageSize);
+
+            SearchResultsBannerDisplay = new List<NotificationBannerModel>();
+            foreach(Notification result in Notifications) {
+                SearchResultsBannerDisplay.Add(new NotificationBannerModel(result, true));
+            }

```

Comment:
The `foreach` loop here seems a bit excessive when a `Select` could do?
`SearchResultsBannerDispaly = Notifications.Select(notification => new NotificationBannerModel(result, true))`
we could probably make it even nicer by getting rid of the boolean parameter, e.g. by having named constructors/ factory functions:
`SearchResultsBannerDispaly = Notifications.Select(NotificationBannerModel.forSearch)` (or similar)

Also not sure we need both `Notifications` and `SearchResultsBannerDisplay` as members on the class?

Let's sit down together for a few minutes to hash it out?